### ISSUE TITLE
Disallow global 

### ISSUE THREAD

**Issue by netei** [eslint/eslint#4488]
I would like to be able to disallow global variables, for example `Promise` , which I don't want to use because we use Bluebird ( with `var P = require("bluebird");`)

Is it somehow possible to disallow the use of a global ?

Because the Promise global is in the buitlin part of https://github.com/sindresorhus/globals/blob/master/globals.json , I do not think it is possible to remove it.

I can see that they are multiple requests about specific rules for each kind of global : Here's a list : 

* Disallow process.env https://github.com/eslint/eslint/issues/657
* Disallow console https://github.com/eslint/eslint/blob/c7ec8a589098f18abd6b1c172334c0c0372aa1fe/docs/rules/no-console.md
*  Feature request: Disable certain globals #2657 
* https://github.com/eslint/eslint/issues/2657#issuecomment-156443036

I think it would make sense to add a rule that is configurable so that you can disallow any globals you want.

Would you accept such a rule ?

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by netei** [comment]
It would maybe even make sense to put this in the core of eslint (since we already have a `globals` key in the config, maybe we could write the following instead : 

"globals": {
  "Promise": 2
}

**Comment by nzakas** [comment]
See our guide on proposing new rules: http://eslint.org/docs/developer-guide/contributing/new-rules

**Comment by timruffles** [comment]
@netei definitely a useful rule. Lots of people will be using a custom promise, and it's a *horrible* pain when you use the global promise instead.

The behaviour required would be disallow using the global variable:

```javascript
var x = Promise.resolve(x); // error - no-use-of-global-promise
```
& allowing creating shadowing/overwriting variables:

```javascript
var Promise = require("bluebird"); // ok
```

**Comment by nzakas** [comment]
We are still looking for a proposal for this rule. Please see the link referenced above.

**Comment by nzakas** [comment]
Duplicate of https://github.com/eslint/eslint/issues/3966
